components/scream/src/share/atm_process/atmosphere_process.hpp

```diff
@@ -80,14 +80,25 @@ class AtmosphereProcess : public ekat::util::enable_shared_from_this<AtmosphereP
   //     be running at the same time, or whether each process can assume
   //     that no other process is currently inside a call to 'run'.
   //   - the finalize method makes sure, if necessary, that all resources are freed.
+  // NOTE: You don't override these methods. Override the protected methods
+  // NOTE: initialize_, run_, and finalize_ instead.
   // The initialize/finalize method should be called just once per simulation (should
   // we enforce that? It depends on what resources we init/free, and how), while the
   // run method can (and usually will) be called multiple times.
   // We should put asserts to verify that the process has been init-ed, when
   // run/finalize is called.
-  virtual void initialize (const TimeStamp& t0) = 0;
-  virtual void run        (const Real dt) = 0;
-  virtual void finalize   (/* what inputs? */) = 0;
+  void initialize (const TimeStamp& t0) {
+    t_ = t0;
+    initialize_(t_);
```

Comment:
I think `blah_` is very close to `blah`, and might be confusing. The field setter methods in this class use the `blah` vs `blah_impl` naming, so maybe we can stick with that convention?